components/x-gift-article/src/v2/SocialShareButtons.jsx

```diff
@@ -0,0 +1,107 @@
+import { h } from '@financial-times/x-engine'
+import { ShareType } from '../lib/constants'
+import oShare from '@financial-times/o-share/main'
+
+new oShare(document.querySelector('[data-o-component=o-share]')) // initialise oShare
```

Comment:
It is strange that you are initializing this code outside the component. Also, using the selector ```[data-o-component=o-share]``` might create some conflict with the share components that are rendered by next-article. I'd recommend moving this code to an `useEffect` when the component is rendered and use `useRef` to select the component instead of a querySelector

Reply:
I previously ran into some issues in `next-article` when `x-gift-article` used the `useState` hook.
Not sure how the renderer works but it seems like `next-article` doesn't play nicely with react hooks.
And now when I made the oShare initialisation changes using `useRef` and `useEffect`, the following error appears on the `next-article` console:
```
Uncaught (in promise) TypeError: react.useRef is not a function
```

snippet 👇🏾 
```jsx
import oShare from '@financial-times/o-share/main'

export const SocialShareButtons = ({ actions, mailtoUrl, mobileShareLinks, shareType }) => {
	const ref = useRef()

	useEffect(() => {
		;(async () => {
			new oShare(ref.current)
		})()
	}, [])
```

Reply:
Using `init()` didn't work as expected

Reply:
Reverted back to initialising `oShare` outside the component. 

Reason for initialising outside the component is to initialise only once and to stop it from initialising every time the component re-renders on state change.

Initialising inside the component results in multiple popups appearing when user clicks on a social share button.
See below for illustration:

https://github.com/Financial-Times/x-dash/assets/10318770/0b5534fc-c76f-464a-a933-235a3c348dd7

Reply:
@fabiosikansi are you happy with this?

Reply:
sorry, I still think that initialising the component outside its lifecycle is a big inconsistency, we cannot predict what would happen. And I also think the selector is too broad and would pick up components in next-article

@abshirahmed the right think would be to init when the component render and destroy it when the component is unmounted, you could try something like:
```
useEffect(() => {
  const oShareInstance = new oShare(ref);
  return () => {
    oShareInstance.destroy();
  };
}, []);
```

this solution removes the origami listeners when the component unmount.

Reply:
@fabiosikansi @jkerr321 [here](https://github.com/Financial-Times/x-dash/compare/5d89ae4b7cfdf5858b423ecef1f1f4a8993da9f0..d31e78decf88d5fa2d31bbb7cd21ed5453e4a5f1) are the latest changes as discusses.
- tested on `next-article` locally 👍🏾